pkg/persistence/db.go

```diff
@@ -163,12 +163,17 @@ func (c *conn) NewKeyClaim(region, originator, hashID string) (string, error) {
 		}
 
 		oneTimeCode := fmt.Sprintf("%08d", n)
-
-		err = persistEncryptionKey(c.db, region, originator, hashID, pub, priv, oneTimeCode)
+		if len(hashID) == 128 {
+			err = persistEncryptionKeyWithHashID(c.db, region, originator, hashID, pub, priv, oneTimeCode)
+		} else {
+			err = persistEncryptionKey(c.db, region, originator, pub, priv, oneTimeCode)
```

Comment:
Curious if there isn't any reason why we couldn't just simplify this, or the parent call slightly, and generate our own hashID if there isn't one being passed in by the caller, and then return it.

From what I gather, we're going to need this pattern for the portal team anyway, and honestly, it's how I would have preferred we rolled with the initial integration.

Reply:
We could - it puts the burden of storing the HashID on the client end though if it is not pre-determined and you would have to build another end point to validate if the OTC has been used.

Reply:
I was thinking just mimicing the behaviour of the withHashID path...

keyclaim -> returns OTC||randomHashIDweGenerate
keyclaim/{hashid} -> does what ever it does now.

Only distinctinction being, we nudge people into only ever calling keyclaim bare path once.